File: Basic/20.Reverse the words in a sentence without reversing the letters/Solution.java
import java.util.Scanner;

public class ReverseWords {
    public static String reverseWords(String sentence) {
        String[] words = sentence.split(" ");
        StringBuilder reversed = new StringBuilder();

        for (int i = words.length - 1; i >= 0; i--) {
            reversed.append(words[i]).append(" ");
        }

        return reversed.toString().trim();
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a sentence: ");
        String sentence = scanner.nextLine();

        String result = reverseWords(sentence);
        System.out.println("Reversed sentence: " + result);
    }
}

File: Basic/20.Reverse the words in a sentence without reversing the letters/Solution.c
#include <stdio.h>
#include <string.h>

void reverseWords(char str[]) {
    int start = 0, end = 0, n = strlen(str);
    char result[1000] = "";

    for (int i = n - 1; i >= 0; i--) {
        if (str[i] == ' ' || i == 0) {
            start = (i == 0) ? 0 : i + 1;
            strncat(result, &str[start], end - start + 1);
            strcat(result, " ");
            end = i - 1;
        }
    }

    printf("Reversed sentence: %s\n", result);
}

int main() {
    char str[100];
    printf("Enter a sentence: ");
    fgets(str, 100, stdin);
    str[strcspn(str, "\n")] = '\0'; // Remove trailing newline

    reverseWords(str);
    return 0;
}
